core/presentation/designsystem/src/main/java/com/nolawiworkineh/designsystem/components/TaskyCalendarRow.kt

```diff
@@ -0,0 +1,87 @@
+package com.nolawiworkineh.designsystem.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.nolawiworkineh.designsystem.Theme.TaskyBlack
+import com.nolawiworkineh.designsystem.Theme.TaskyGray
+import com.nolawiworkineh.designsystem.Theme.TaskyTheme
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.temporal.TemporalAdjusters
+
+@Composable
+fun CalendarDaysRow(
+    selectedDate: LocalDate,
+    onDateSelected: (LocalDate) -> Unit,
+    modifier: Modifier = Modifier
+) {
+
+    val sundayOfWeek = selectedDate.with(TemporalAdjusters.previousOrSame(DayOfWeek.SUNDAY))
```

Comment:
`sundayOfWeek` will be re-initialized during each recomposition, which can happen very frequently. With the `remember` block, you can persist this value across recompositions.

```
val sundayWeek by remember(key = selectedDate) {
  mutableStateOf(
    selectedDate.with(TemporalAdjusters.previousOrSame(DayOfWeek.SUNDAY))
  )
}
```

This won’t, however, protect you against configuration changes (device orientation change, dark mode toggle). That is why I’d strongly recommend putting all UI states in a ViewModel. You can then calculate the days you want to display and then update your state which in turn will update your UI.